Add Sundials 5.2.0 & 5.3.0

vvilpas opened the pull request:
Specify library name and version:  **sundials/5.2.0 & 5.3.0** 

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

uilianries commented on recipes/sundials/all/conanfile.py:
```diff
@@ -0,0 +1,85 @@
+
+class SundialsConan(ConanFile):
+    name = "sundials"
+    license = "BSD 3-Clause"
```
```suggestion
    license = "BSD-3-Clause"
```

uilianries commented on recipes/sundials/all/conanfile.py:
```diff
@@ -0,0 +1,85 @@
+               "build_idas": [True, False],
+               "build_kinsol": [True, False]}
+    default_options = {"shared": False,
+                       "fPIC": False,
```
```suggestion
                       "fPIC": True,
```

uilianries commented on recipes/sundials/all/conanfile.py:
```diff
@@ -0,0 +1,85 @@
+                       "build_kinsol": True}
+    exports_sources = "CMakeLists.txt", "patches/**"
+    generators = "cmake"
+    short_paths = True
```
```suggestion
```

vvilpas replied:
It doesn't compile in windows without this, due to very long names.

uilianries replied:
fair enough

uilianries commented on recipes/sundials/all/conanfile.py:
```diff
@@ -0,0 +1,85 @@
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_SHARED_LIBS"] = self.options.shared
```
```suggestion
```

Automatically configured by CMake helper

vvilpas commented:
I have removed the tests and put a minimum example to test some includes and linking.
Addressed all other comments except short_paths, beacuse it doesn't compile in windows due to long names issue.
